test/demo/utils/fdc3RemoteExecution.ts

```diff
@@ -103,17 +100,25 @@ export async function addContextListener(executionTarget: Identity): Promise<Rem
     return createRemoteContextListener(executionTarget, id);
 }
 
-export async function addIntentListener(executionTarget: Identity, intent: IntentType): Promise<RemoteIntentListener> {
-    const id = await ofBrowser.executeOnWindow(executionTarget, function (this: TestWindowContext, intentRemote: IntentType): number {
+export async function addIntentListener(executionTarget: Identity, intent: IntentType, listener?: (context: Context) => any): Promise<RemoteIntentListener> {
+    const remoteFn = listener ? await ofBrowser.getOrMountRemoteFunction(executionTarget, listener) : undefined;
+    const id = await ofBrowser.executeOnWindow(executionTarget, function (this: TestWindowContext, intentRemote: IntentType, listenerRemote?: any): number {
```

Comment:
`listenerRemote` could be typed as `(context: Context) => Promise<any>`, and then `remoteFn` cast to that as it's passed into `executeOnWindow`.

I guess behind the scenes a `JSHandle` is turned into an actual function, but that's not properly captured by the type system.

It's messy, so I'm marginal on this, fine with you saying you prefer as is

Reply:
Typed `remoteFn` into a behemoth.

Reply:
I think as is, the cast is premature, and should happen as `remoteFn` is passed into `executeOnWindow` on line 128, rather than on line 106. Until it's gone through `executeOnWindow`, `remoteFn` is still unambiguously a `JSHandler` and not a callable function, so the cast should go later.